python/tank/util/shotgun.py

```diff
@@ -224,24 +224,59 @@ def download_url(sg, url, location):
     :param location: path on disk where the payload should be written.
                      this path needs to exists and the current user needs
                      to have write permissions
+    :param bool use_url_extension: Optionally replace the file extension of the
```

Comment:
I still feel there is a concrete example missing. Maybe we could add something like the follwing to the main docstring, before the params?

```
In some cases, the target content of the url is not known on beforehand. For example, 
the url ``https://my-site.shotgunstudio.com/thumbnail/full/Asset/1227`` may resolve into
a ``https://some-site/path/to/a/thumbnail.png``. In such cases, you can set the 
use_url_extension parameter to True - this will cause the method to append the
file extension of the resolved url to the filename passed in via the location parameter. 
So for the urls given above, you would get the following results:

- location="/path/to/file" and use_url_extension=False would return "/path/to/file"
- location="/path/to/file" and use_url_extension=True would return "/path/to/file.png"

```